plugin/veracode_static_scan.go

```diff
@@ -0,0 +1,299 @@
+
+	timeout := time.After(time.Duration(args.Timeout) * time.Minute)
+	ticker := time.NewTicker(1 * time.Minute)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-timeout:
+			if args.TimeoutFailsJob {
+				return fmt.Errorf("❌ Scan polling timed out after %d minutes", args.Timeout)
+			}
+			logrus.Warn("⚠️ Scan polling timed out, but job not marked as failed")
+			return nil
+
+		case <-ticker.C:
+			logrus.Info("🔁 Checking scan status...")
+
+			published, policyStatus, err := isScanPublished(ctx, args)
+			if err != nil {
+				logrus.Warnf("⚠️ Failed to check scan status: %v", err)
+				continue
+			}
+			if !published {
+				continue
+			}
+
+			logrus.Infof("✅ Scan completed and published! Policy compliance status: %s", policyStatus)
+
+			if args.CanFailJob {
+				if policyStatus == "Did Not Pass" || policyStatus == "Conditional Pass" {
+					return fmt.Errorf("❌ Policy evaluation returned %s. Marking build as Failed", policyStatus)
+				}
+			}
+
+			return nil
+		}
+	}
+}
+
+func isScanPublished(ctx context.Context, args Args) (bool, string, error) {
+	appID, err := getAppID(ctx, args)
+	if err != nil {
+		return false, "", err
+	}
+
+	statusArgs := []string{
+		"-jar", "/opt/veracode/api-wrapper.jar",
+		"-action", "GetBuildInfo",
+		"-appid", appID,
+		"-vid", args.VID,
+		"-vkey", args.VKey,
+	}
+	if args.SandboxName != "" {
+		statusArgs = append(statusArgs, "-sandboxname", args.SandboxName)
+	}
+
+	masked := maskSensitiveArgs(statusArgs)
+	logrus.Infof("➡️  Checking build info with: java %s", strings.Join(masked, " "))
+
+	cmd := exec.CommandContext(ctx, "java", statusArgs...)
+	output, err := cmd.CombinedOutput()
+	outputStr := string(output)
+	logrus.Infof("🔍 CLI raw output:\n%s", outputStr)
+
+	if err != nil {
+		return false, "", fmt.Errorf("failed to check build info: %w", err)
+	}
+
+	isPublished := strings.Contains(outputStr, "published") || strings.Contains(outputStr, "Results Ready")
+	policyStatus := extractPolicyComplianceStatus(outputStr)
+
+	return isPublished, policyStatus, nil
+}
+
+func getAppID(ctx context.Context, args Args) (string, error) {
+	appListArgs := []string{
+		"-jar", "/opt/veracode/api-wrapper.jar",
+		"-action", "GetAppList",
+		"-vid", args.VID,
+		"-vkey", args.VKey,
+	}
+
+	masked := maskSensitiveArgs(appListArgs)
+	logrus.Infof("➡️  Fetching App ID with: java %s", strings.Join(masked, " "))
+
+	cmd := exec.CommandContext(ctx, "java", appListArgs...)
+	output, err := cmd.CombinedOutput()
+	outputStr := string(output)
+	logrus.Infof("🔍 AppList raw output:\n%s", outputStr)
+
+	if err != nil {
+		return "", fmt.Errorf("failed to get app list: %w", err)
+	}
+
+	appID := extractAppID(outputStr, args.AppName)
+	if appID == "" {
+		return "", fmt.Errorf("app name '%s' not found in app list", args.AppName)
+	}
+
+	logrus.Infof("✅ Found App ID for '%s': %s", args.AppName, appID)
+	return appID, nil
+}
+
+func extractAppID(xmlStr, appName string) string {
```

Comment:
use a proper XML parsing library to handle the XML responses rather than string manipulation and address the same in `extractPolicyComplianceStatus()`